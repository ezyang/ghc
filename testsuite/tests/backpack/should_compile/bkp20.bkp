{-# LANGUAGE TypeFamilies #-}

unit p where
    signature H where
        type family Elem c

unit q where
    signature H where
        type family Elem c

unit r where
    include p
    include q
    module M where
        import H
        type instance Elem Bool = Int

unit h-impl where
    module H where
        type family Elem c
unit s where
    include h-impl
    include r
