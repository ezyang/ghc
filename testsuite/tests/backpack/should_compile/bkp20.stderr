[1 of 5] Processing p
/--- Shape for p
provides:
requires: H -> Elem{Elem}
\---
  [1 of 1] Compiling H[sig]           ( p/H.hsig, nothing )
[2 of 5] Processing q
/--- Shape for q
provides:
requires: H -> Elem{Elem}
\---
  [1 of 1] Compiling H[sig]           ( q/H.hsig, nothing )
[3 of 5] Processing r
/--- Shape for r
provides: M -> r(hole:H):M
requires: H -> Elem{Elem}
\---
  [1 of 2] Compiling H[sig]           ( r/H.hsig, nothing )
  [2 of 2] Compiling M                ( r/M.hs, nothing )
[4 of 5] Processing h-impl
  Instantiating h-impl
  [1 of 1] Compiling H                ( h-impl/H.hs, bkp20-out/h-impl/H.o )
[5 of 5] Processing s
  Instantiating s
  [1 of 2] Including h-impl
  [2 of 2] Including r
    Instantiating r(h-impl:H)
    [1 of 2] Including q
      Instantiating q(h-impl:H)
      [1 of 1] Compiling H[sig]           ( q/H.hsig, nothing )
    [2 of 2] Including p
      Instantiating p(h-impl:H)
      [1 of 1] Compiling H[sig]           ( p/H.hsig, nothing )
    [1 of 2] Compiling H[sig]           ( r/H.hsig, nothing )
    [2 of 2] Compiling M                ( r/M.hs, bkp20-out/r/r-5FYQgnNkfSvBT5yogOxPpf/M.o )
